### PR TITLE
Complex Unlocking Criteria

### PR DESCRIPTION
**Pull request by Bentechy66**
Switches out the `unlocks` field for a `unlock_requirements` field, which takes Reverse Polish Notation(tm) of challenge IDs, supporting AND and OR queries.

### CODE REVIEW

**Comment by david-cooke** [comment]
ractf - now with turing complete challenge unlocks

**Review by thebeanogamer — COMMENTED** [review]
0/10, not enough use of `exec()`

**Comment by david-cooke** [comment]
see earlier comment, no stack based vms except java are allowed to run on ix420

**Comment by Bentechy66** [comment]
ayayay

**Review by david-cooke — CHANGES_REQUESTED** [review]
For this to be production ready, its going to need support for XOR, NAND, NOR and XNOR.

**Comment by david-cooke** [comment]
I would also like to see a more versatile instruction set however once the bare minimum is covered, we can figure the rest out from there

**Comment by david-cooke** [comment]
tl;dr if it isnt an rce vuln its not powerful enough

**Comment by david-cooke** [comment]
you literally added lootboxes to the api then refuse to add xor to challenge unlocking

**Review by 404dcd — APPROVED** [review]
LGTM

**Review by AmusedNetwork — CHANGES_REQUESTED** [review]
Change this commit title and we can get it merged.

**Review by david-cooke — APPROVED** [review]
LGTM
